specviz/core/items.py

```diff
@@ -142,7 +143,7 @@ def is_data_unit_compatible(self, unit):
         return (self.data_item.flux.unit == "" or
                 unit is not None and
                 self.data_item.flux.unit.is_equivalent(
-                    unit, equivalencies=spectral_density(self.spectral_axis)))
+                    unit, equivalencies=spectral_density(self.spectral_axis*u.Unit(self.spectral_axis_unit))))
```

Comment:
```suggestion
                    unit, equivalencies=spectral_density(self.data_item.spectral_axis)))
```

Functionally equivalent, but a little cleaner.

Reply:
Looking at the CI test, it failed because it stalled out when installing pyqt from pip -- I don't think that's related.